thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/HashService.js

```diff
@@ -42,11 +42,19 @@ HashService.prototype = {
     switch (key) {
       case HASH_PARAMS.ANOMALIES_START_DATE:
       case HASH_PARAMS.ANOMALIES_END_DATE:
+      case HASH_PARAMS.ANALYSIS_CURRENT_START:
+      case HASH_PARAMS.ANALYSIS_CURRENT_END:
+      case HASH_PARAMS.ANALYSIS_BASELINE_START:
+      case HASH_PARAMS.ANALYSIS_BASELINE_END:
         value = moment(value);
         break;
       case HASH_PARAMS.ANOMALIES_PAGE_NUMBER:
         value = Number(value);
         break;
+      case HASH_PARAMS.ANALYSIS_FILTERS:
+        if (typeof value === "string") {
```

Comment:
Hmm.. this may not actually achieve what you want... `JSON.parse("")` will throw. Personally, if i'm uncertain of the input into `JSON.parse` I would encase in a try-catch block then subsequently check that value is valid